### PR TITLE
Register screen - Date Picker

### PR DESCRIPTION
**Pull request by MirzadS**

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -46,20 +40,18 @@ class _MyHomePageState extends State<MyHomePage> {
...
-          children: <Widget>[
-            const Text(
-              'You have pushed the button this many times:',
-            ),
-            Text(
-              '$_counter',
-              style: Theme.of(context).textTheme.headline4,
-            ),
-          ],
+          // children: [],
```
Comment (DenisAPPIT) [inline lib/main.dart]: remove comment

File: lib/main.dart [hunk 2]
```diff
@@ -46,20 +40,18 @@ class _MyHomePageState extends State<MyHomePage> {
...
         ),
       ),
       floatingActionButton: FloatingActionButton(
-        onPressed: _incrementCounter,
-        tooltip: 'Increment',
+        onPressed: () {
+          Navigator.of(context).push(
+            MaterialPageRoute(
+              // builder: (context) => const RegisterForm(),
```
Comment (DenisAPPIT) [inline lib/main.dart]: remove comment

File: lib/ui/Helper/data_picker.dart [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  const DatePicker({Key? key, required this.date}) : super(key: key);
+
+  final date;
+
+  @override
+  State<DatePicker> createState() => _DatePickerState();
+}
+
+class _DatePickerState extends State<DatePicker> {
+  // TextEditingController _date = TextEditingController();
```
Comment (DenisAPPIT) [inline lib/ui/Helper/data_picker.dart]: remove comment

File: lib/ui/Helper/data_picker.dart [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text('DatePicker on TextField'),
+      ),
+      body: Container(
+        margin: const EdgeInsets.only(left: 16, right: 32),
+        child: TextField(
+            controller: widget.date,
+            // obscureText: true,
```
Comment (DenisAPPIT) [inline lib/ui/Helper/data_picker.dart]: remove

File: lib/ui/Helper/data_picker.dart [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+              icon: Icon(Icons.date_range),
+              hintText: "Select Date",
+            ),
+            readOnly: true,
+            onTap: () async {
+              DateTime? pickedDate = await showDatePicker(
+                  context: context,
+                  initialDate: DateTime.now(),
+                  firstDate: DateTime(
+                      2000), //DateTime.now() - not to allow to choose before today.
```
Comment (DenisAPPIT) [inline lib/ui/Helper/data_picker.dart]: remove comment

File: lib/ui/Helper/data_picker.dart [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+                  lastDate: DateTime(2101));
+
+              if (pickedDate != null) {
+                setState(() {
+                  widget.date.text = DateFormat('yyyy-MM-dd')
+                      .format(pickedDate); //set output date to TextField value.
```
Comment (DenisAPPIT) [inline lib/ui/Helper/data_picker.dart]: remove comment

File: lib/ui/Helper/data_picker.dart (hunk 5 above)
Comment (DenisAPPIT) [reply lib/ui/Helper/data_picker.dart]: why DateTime(2000), what will this do?

File: lib/ui/background/background.dart [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+      ),
+    );
+  }
+}
+
+class Curved extends CustomPainter {
+  @override
+  void paint(Canvas canvas, Size size) {
+    var rect = Offset.zero & size;
+    // Path rectPathThree = Path();
```
Comment (DenisAPPIT) [inline lib/ui/background/background.dart]: remove

File: lib/ui/background/background.dart [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+      size.height * 0.18,
+    );
+    path.quadraticBezierTo(
+      size.width * 0.06,
+      size.height * 0.21,
+      size.width * 0,
+      size.height * 0.26,
+    );
+    path.close();
+    //
```
Comment (DenisAPPIT) [inline lib/ui/background/background.dart]: remove

File: lib/ui/forms/register_form.dart [hunk 9]
```diff
@@ -0,0 +1,197 @@
...
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: Stack(children: [
+        Background(),
+        Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: [
+            // Background(),
```
Comment (DenisAPPIT) [inline lib/ui/forms/register_form.dart]: remove comment

File: lib/ui/forms/register_form.dart [hunk 10]
```diff
@@ -0,0 +1,197 @@
...
+
+                                if (pickedDate != null) {
+                                  setState(() {
+                                    _date.text = DateFormat('yyyy-MM-dd')
+                                        .format(pickedDate);
+                                  });
+                                }
+                              }),
+                        ),
+                        // DatePicker(date: _date)
```
Comment (DenisAPPIT) [inline lib/ui/forms/register_form.dart]: remove

File: lib/ui/forms/register_form.dart [hunk 11]
```diff
@@ -0,0 +1,197 @@
...
+                      ],
+                    ),
+                  ),
+
+                  // KRUG
```
Comment (DenisAPPIT) [inline lib/ui/forms/register_form.dart]: remove

**Review by DenisAPPIT — CHANGES_REQUESTED** [review]

File: lib/ui/Helper/data_picker.dart (hunk 5 above)
Comment (MirzadS) [reply lib/ui/Helper/data_picker.dart]: Changed to 1900

Users will be able to choose a year between 1900 and now

**Pull request merged**